react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +107,109 @@ describe('getDataPushdata', () => {
-    expect(() =>
-      getDataPushdata(data206bytes)
-    ).toThrow('Maximum 205 byte size exceeded for user data: 206')
-   })
-})
+    const data206bytes = '%'.repeat(206);
+    expect(() => getDataPushdata(data206bytes)).toThrow(
+      'Maximum 205 byte size exceeded for user data: 206',
+    );
+  });
+});
 
-describe('parseOpReturnProps', () => {
+describe('encodeOpReturnProps', () => {
   // protocol pushdata + protocol + version byte
-  const allResultsPrefix = '04' + '50415900' + '00'
-  const paymentIdDigits = 18
-  const paymentIdRegex = /^[0-9a-fA-F]{18}$/
-  it('Undefined data', () => {
-    const fullResult = parseOpReturnProps(undefined)
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+  const allResultsPrefix = '04' + '50415900' + '00';
+  const paymentIdDigits = 18;
+  const paymentIdRegex = /^[0-9a-fA-F]{18}$/;
+  it('Parses undefined data', () => {
+    const fullResult = encodeOpReturnProps(undefined);
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(allResultsPrefix + '00'); // 00 is pushdata for no data
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Undefined data no paymentId', () => {
-    const fullResult = parseOpReturnProps(undefined, true)
-    expect(fullResult).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-  })
+    const fullResult = encodeOpReturnProps(undefined, true);
+    expect(fullResult).toBe(allResultsPrefix + '00' + '00'); // the 00s are pushdata for the data and paymentId respectively
+  });
   it('Empty data', () => {
-    const fullResult = parseOpReturnProps('')
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = encodeOpReturnProps('');
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(allResultsPrefix + '00'); // 00 is pushdata for no data
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Empty data no paymentId', () => {
-    const fullResult = parseOpReturnProps('', true)
-    expect(fullResult).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-  })
+    const fullResult = encodeOpReturnProps('', true);
+    expect(fullResult).toBe(allResultsPrefix + '00' + '00'); // the 00s are pushdata for the data and paymentId respectively
+  });
   it('Simple parse OpReturn 1', () => {
```

Comment:
Need to move away from these numbered test names. The value of the unit tests is not in the quantity of tests, but in covering every expected case.

If there are 19 cases, we need 19 tests. If there are 5 cases, 5 tests is great.

Here we are testing `it('Encodes a PayButton v0 protocol transaction with custom user data and nonce')`